test/recipes/06-test_algorithmid.t

```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/perl
+
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+# 
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+
+use OpenSSL::Test qw(:DEFAULT srctop_file);
+use OpenSSL::Test::Utils;
+
+setup("test_algorithmid");
+
+# eecert => cacert
+my %certs_info =
+    (
+     'ee-cert' => 'ca-cert',
+     'ee-cert2' => 'ca-cert2',
+     'ee-cert-ec-explicit' => 'ca-cert-ec-named',
+     'ee-cert-ec-named-explicit' => 'ca-cert-ec-explicit',
+     'ee-cert-ec-named-named' => 'ca-cert-ec-named',
+     # 'ee-pss-sha1-cert' => 'ca-cert',
+     # 'ee-pss-sha256-cert' => 'ca-cert',
+     # 'ee-pss-cert' => 'ca-pss-cert',
+     'server-ecdsa-brainpoolP256r1-cert' => 'rootcert',
+     'server-pss-restrict-cert' => 'rootcert',
+     # 'server-ed448-cert' => 'root-ed448-cert'
```

Comment:
The reason some of these are currently commented away is that `EVP_DigestVerifyInit()` fails for ED448, and that there's a mismatch between the RSASSA-PSS signature in ee-pss certs and the acceptable signature algorithm given from the EVP_KEYMGMT...  That needs deeper coding to resolve, and wll *not* be part of this PR.